src/VatRates/TaxesEuropeDatabaseClient.php

```diff
@@ -0,0 +1,93 @@
+<?php declare(strict_types = 1);
+
+namespace SandwaveIo\Vat\VatRates;
+
+use DateTime;
+use SoapClient;
+use SoapFault;
+
+final class TaxesEuropeDatabaseClient implements ResolvesVatRates
+{
+    const WSDL = 'https://ec.europa.eu/taxation_customs/tedb/ws/VatRetrievalService.wsdl';
+    const SOAP_CONFIG = [
+        'version' => SOAP_1_1,
+        'cache_wsdl' => WSDL_CACHE_MEMORY,
+    ];
+    /* @see https://ec.europa.eu/taxation_customs/tedb/ws/VatRetrievalServiceType.xsd */
+    const RATE_TYPE_STANDARD = 'STANDARD';
+    const RATE_VALUE_TYPE_DEFAULT = 'DEFAULT';
+
+    private SoapClient $client;
+
+    public function __construct(?SoapClient $client = null)
+    {
+        $this->client = $client ?? new SoapClient(self::WSDL, self::SOAP_CONFIG);
+    }
+
+    /**
+     * Retrieve the default VAT rate for a given country. If the countries VAT rate cannot be resolved, null is returned.
+     * This also happens if the country does not exist in the European Tax Database.
+     *
+     * @param string        $countryCode ISO country code
+     * @param DateTime|null $dateTime    Date of checking, defaults to today.
+     *
+     * @return float|null
+     */
+    public function getDefaultVatRateForCountry(string $countryCode, ?DateTime $dateTime = null): ?float
+    {
+        $date = $dateTime ?? new DateTime();
+        $response = $this->call('retrieveVatRates', [
+            'retrieveVatRatesRequest' => [
+                'memberStates' => [$countryCode],
+                'situationOn' => $date->format('Y-m-d'),
+            ],
+        ]);
```

Comment:
Why not just $this->client->retrieveVatRates(....)

Probably saves a lot of code.

Reply:
Maybe we should, I'll take a look

Reply:
On second thought, I don't think I want to do this. I find it confusing that IDE's cannot index the client correctly (based on the WSDL). On top of that we would have to add PHPStan exceptions for this. With this implementation, it is clear that a certain call is made, and the error handling for the call is moved to a dedicated function.

Reply:
I am resolving this for now, we can refactor it later when the VIES API implementation is added.